elastica/modules/forcing.py

```diff
@@ -54,37 +57,37 @@ def _finalize_forcing(self):
         # inplace : https://stackoverflow.com/a/1208792
 
         # dev : the first index stores the rod index to apply the boundary condition
-        # to. Technically we can use another array but it its one more book-keeping
-        # step. Being lazy, I put them both in the same array
-        self._ext_forces_torques[:] = [
-            (ext_force_torque.id(), ext_force_torque())
-            for ext_force_torque in self._ext_forces_torques
-        ]
-
-        # Sort from lowest id to highest id for potentially better memory access
-        # _ext_forces_torques contains list of tuples. First element of tuple is
-        # rod number and following elements are the type of boundary condition such as
-        # [(0, NoForces, GravityForces), (1, UniformTorques), ... ]
-        # Thus using lambda we iterate over the list of tuples and use rod number (x[0])
-        # to sort _ext_forces_torques.
-        self._ext_forces_torques.sort(key=lambda x: x[0])
-
-        # Find if there are any friction plane forcing, if add them to the end of the list,
-        # since friction planes uses external forces.
-        friction_plane_index = []
-        for idx, ext_force_torque in enumerate(self._ext_forces_torques):
-            if isinstance(ext_force_torque[1], AnisotropicFrictionalPlane):
-                friction_plane_index.append(idx)
-
-        # Move to the friction forces to the end of the external force and torques list.
-        for index in friction_plane_index:
-            self._ext_forces_torques.append(self._ext_forces_torques.pop(index))
-
-    def _call_ext_forces_torques(self, time, *args, **kwargs):
-        for sys_id, ext_force_torque in self._ext_forces_torques:
-            ext_force_torque.apply_forces(self._systems[sys_id], time, *args, **kwargs)
-            ext_force_torque.apply_torques(self._systems[sys_id], time, *args, **kwargs)
-            # TODO Apply torque, see if necessary
+        # to.
+        for external_force_and_torque in self._ext_forces_torques:
+            sys_id = external_force_and_torque.id()
+            forcing_instance = external_force_and_torque.instantiate()
+
+            apply_forces = functools.partial(
+                forcing_instance.apply_forces, system=self._systems[sys_id]
+            )
+            apply_torques = functools.partial(
+                forcing_instance.apply_torques, system=self._systems[sys_id]
+            )
+
+            self._feature_group_synchronize.add_operators(
+                external_force_and_torque, [apply_forces, apply_torques]
+            )
+
+            self.warnings(external_force_and_torque)
+
+        self._ext_forces_torques = []
+        del self._ext_forces_torques
+
+    def warnings(self, external_force_and_torque):
+        from elastica.interaction import AnisotropicFrictionalPlane
```

Comment:
We moved to friction plane under contact. Can you change accordingly.

Reply:
I see `AnisotropicFrictionalPlane` in `interaction.py`:
https://github.com/GazzolaLab/PyElastica/blob/e71e54790d5b782ad1a2d9856888b09783251fd0/elastica/interaction.py#L158C7-L158C33

Reply:
I removed warning for AnisotropicFrictionalPlane. @Ali-7800 will remove the class.